pkg/util/dynamichelper/createorder.go

```diff
@@ -4,40 +4,36 @@ package dynamichelper
 // Licensed under the Apache License 2.0.
 
 import (
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"reflect"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	extensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// rather than list every GK, just list the ones whose creation really has to be
-// brought forward
-var createOrderMap = map[string]int{
+// rather than list every object, just list the ones whose creation really has
+// to be brought forward
+var createOrderMap = map[reflect.Type]int{
 	// non-namespaced resources
-	"CustomResourceDefinition.apiextensions.k8s.io":    1, // before custom resources
-	"ClusterRole.rbac.authorization.k8s.io":            2, // before workload resources
-	"ClusterRoleBinding.rbac.authorization.k8s.io":     3, // before workload resources
-	"SecurityContextConstraints.security.openshift.io": 4, // before workload resources
+	reflect.TypeOf(&extensionsv1beta1.CustomResourceDefinition{}): -8, // before custom resources
+	reflect.TypeOf(&rbacv1.ClusterRole{}):                         -7, // before workload resources
+	reflect.TypeOf(&rbacv1.ClusterRoleBinding{}):                  -6, // before workload resources
+	reflect.TypeOf(&securityv1.SecurityContextConstraints{}):      -5, // before workload resources
```

Comment:
Why you use negative values? want to understand thinking :)

Reply:
I guess this is to ensure that any key not in the map returns the zero value for int, which will be greater than any value of an already known type in the map. And hence will be done at the end of the order. This is more "concise" and "elegant" than previous code.
Let's see if I guessed right :)

Reply:
Yes, you did ;-)